lang/gcc7/Portfile

```diff
@@ -30,13 +30,30 @@ master_sites        ftp://ftp.funet.fi/pub/mirrors/sources.redhat.com/pub/gcc/re
                     ftp://gcc.gnu.org/pub/gcc/releases/gcc-${version}/ \
                     gnu:gcc/gcc-${version}
 
-distname            gcc-${version}
-use_xz              yes
 
 checksums           rmd160  91d46ec088badec75f41a2ad2a0ba228a6715107 \
                     sha256  b81946e7f01f90528a1f7352ab08cc602b9ccc05d4e44da4bd501c5a189ee661 \
                     size    62783088
 
+if { ${os.platform} eq "darwin" && ${cxx_stdlib} ne "libc++" } {
```

Comment:
the test could be for the os version to be < 10.6, as that would cover off most of the wreckage with 7.5.0. But I haven't got any systems running libstdc++ with 10.6 to 10.8 any more, and I suspect that those systems will probably also have this ABI issue...so that's why I did it this way instead.